src/components/Table/Table.tsx

```diff
@@ -170,6 +175,138 @@ function InternalTable<RecordType extends object = any>(
+    );
+    const [cancelSortText, setCancelSortText] = useState<string>(
+        defaultCancelSortText
+    );
+
+    // Memoized locs, if the prop isn't provided use the loc defaults.
+    useMemo(() => {
+        setFilterConfirmText(
+            props.filterConfirmText
+                ? props.filterConfirmText
+                : locale.lang!.filterConfirmText
+        );
+        setFilterResetText(
+            props.filterResetText
+                ? props.filterResetText
+                : locale.lang!.filterResetText
+        );
+        setFilterEmptyText(
+            props.filterEmptyText
+                ? props.filterEmptyText
+                : locale.lang!.filterEmptyText
+        );
+        setFilterCheckallText(
+            props.filterCheckallText
+                ? props.filterCheckallText
+                : locale.lang!.filterCheckallText
+        );
+        setFilterSearchPlaceholderText(
+            props.filterSearchPlaceholderText
+                ? props.filterSearchPlaceholderText
+                : locale.lang!.filterSearchPlaceholderText
+        );
+        setEmptyText(
+            props.emptyText ? props.emptyText : locale.lang!.emptyText
+        );
+        setEmptyTextDetails(
+            props.emptyTextDetails
+                ? props.emptyTextDetails
+                : locale.lang!.emptyTextDetails
+        );
+        setSelectInvertText(
+            props.selectInvertText
+                ? props.selectInvertText
+                : locale.lang!.selectInvertText
+        );
+        setSelectNoneText(
+            props.selectNoneText
+                ? props.selectNoneText
+                : locale.lang!.selectNoneText
+        );
+        setSelectionAllText(
+            props.selectionAllText
+                ? props.selectionAllText
+                : locale.lang!.selectionAllText
+        );
+        setExpandText(
+            props.expandText ? props.expandText : locale.lang!.expandText
+        );
+        setCollapseText(
+            props.collapseText ? props.collapseText : locale.lang!.collapseText
+        );
+        setTriggerDescText(
+            props.triggerDescText
+                ? props.triggerDescText
+                : locale.lang!.triggerDescText
+        );
+        setTriggerAscText(
+            props.triggerAscText
+                ? props.triggerAscText
+                : locale.lang!.triggerAscText
+        );
+        setCancelSortText(
+            props.cancelSortText
+                ? props.cancelSortText
+                : locale.lang!.cancelSortText
+        );
+    }, [
+        filterConfirmText,
+        filterResetText,
+        filterEmptyText,
+        filterCheckallText,
+        filterSearchPlaceholderText,
+        emptyText,
+        emptyTextDetails,
+        selectInvertText,
+        selectNoneText,
+        selectionAllText,
+        expandText,
+        collapseText,
+        triggerDescText,
+        triggerAscText,
+        cancelSortText,
```

Comment:
with the current setup, these won't ever change. Thats probably fine, just pointing it out.

Reply:
Updating dep to locale only as it may change.